src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs

```diff
@@ -99,6 +97,12 @@ public NuGetVersion Version
         public RuntimeGraph RuntimeGraph { get; set; } = new RuntimeGraph();
 
         /// <summary>
+        /// Project Settings is used to pass settings like HideWarningsAndErrors down to lower levels.
+        /// This should not be part of the Equals and GetHashCode.
+        /// </summary>
+        public IDictionary<string, object> ProjectSettings { get; set; } = new Dictionary<string, object>();
```

Comment:
I like this idea, but since there is currently just 1 bool I think this should be a new class with only that property as a bool.

I also don't see being read/written which means that it will be lost during ``Clone()`` and that is standard for all dg spec operations.
https://github.com/NuGet/NuGet.Client/blob/d95e1fc5fae322cfab69e9d4c025ec49e3febbb1/src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs#L193-L204

When you add support for persisting this I think it will be more obvious that this needs to be a simple data type since you can't persist just any object to a json file.